src/pool/srv_target.c

```diff
@@ -989,19 +989,34 @@ ds_pool_tgt_query_handler(crt_rpc_t *rpc)
 {
 	struct pool_tgt_query_in	*in = crt_req_get(rpc);
 	struct pool_tgt_query_out	*out = crt_reply_get(rpc);
-	struct ds_pool_hdl		*hdl;
+	struct ds_pool_hdl		*hdl = NULL;
+	struct ds_pool			*pool;
 	int				 rc;
 
-	hdl = ds_pool_hdl_lookup(in->tqi_op.pi_hdl);
-	if (hdl == NULL) {
-		D_ERROR("Failed to find pool hdl "DF_UUID"\n",
-			DP_UUID(in->tqi_op.pi_hdl));
-		D_GOTO(out, rc = -DER_NO_HDL);
+	/* If no handle provided, try looking up by pool UUID */
+	if (uuid_is_null(in->tqi_op.pi_hdl)) {
+		pool = ds_pool_lookup(in->tqi_op.pi_uuid);
+		if (pool == NULL) {
+			D_ERROR("Failed to find pool "DF_UUID"\n",
+				DP_UUID(in->tqi_op.pi_uuid));
+			D_GOTO(out, rc = -DER_NONEXIST);
+		}
+	} else {
+		hdl = ds_pool_hdl_lookup(in->tqi_op.pi_hdl);
+		if (hdl == NULL) {
+			D_ERROR("Failed to find pool hdl "DF_UUID"\n",
+				DP_UUID(in->tqi_op.pi_hdl));
+			D_GOTO(out, rc = -DER_NO_HDL);
+		}
+
+		D_ASSERT(hdl->sph_pool != NULL);
+		pool = hdl->sph_pool;
```

Comment:
It appears we could simply remove this branch and use in->tqi_op.pi_uuid in all cases, couldn't we...?

Reply:
We could - I wasn't sure if we wanted to save cycles by using the pool from the handle if we already had it, instead of looking it up. I can do that in a follow-on patch if you like.

Reply:
Cycles should be about the same. "If no handle, use the pool UUID" leads to "why looking at the handle at all". I think the original author might have daos_pool_query_target in mind. Since that's unimplemented, we can't find a use case for handles at the moment. We can leave this as a future security-related task, perhaps.

Reply:
It would be quick enough to go ahead and remove the branch from here. If someone needs the handle in this flow sometime later, they can, of course, re-add it. I'll create a PR.

Reply:
Created PR #1672.